### PR TITLE
Even more documentation for various controllers and libs.

### PR DESCRIPTION
**Pull request by danjm**
Ref #3941 

app/scripts/controllers/balance.js
app/scripts/controllers/blacklist.js
app/scripts/controllers/recent-blocks.js
app/scripts/lib/account-tracker.js
app/scripts/lib/extractEthjsErrorMessage.js
app/scripts/lib/getObjStructure.js
app/scripts/lib/message-manager.js
app/scripts/lib/notification-manager.js
app/scripts/lib/pending-balance-calculator.js
app/scripts/lib/personal-message-manager.js
app/scripts/lib/seed-phrase-verifier.js
app/scripts/lib/typed-message-manager.js

### CODE REVIEW

File: app/scripts/controllers/preferences.js [hunk 1]
```diff
@@ -7,9 +7,9 @@ class PreferencesController {
   /**
    *
    * @typedef {Object} PreferencesController
-   * @param {object} opts Overrides the defaults for the initial state of this.store
-   * @property {object} store The an object containing a users preferences, stored in local storage
-   * @property {array} store.frequentRpcList A list of custom rpcs to provide the user
+	 * @param {object} opts Overrides the defaults for the initial state of this.store
```
Comment (bitpshr) [inline app/scripts/controllers/preferences.js]: This line needs one extra space at the beginning.

File: app/scripts/controllers/preferences.js [hunk 2]
```diff
@@ -7,9 +7,9 @@ class PreferencesController {
...
+   * @property {object} store The stored object containing a users preferences, stored in local storage
+	 * @property {array} store.frequentRpcList A list of custom rpcs to provide the user
```
Comment (bitpshr) [inline app/scripts/controllers/preferences.js]: This line needs one extra space at the beginning.

File: app/scripts/lib/message-manager.js [hunk 3]
```diff
@@ -3,8 +3,45 @@ const ObservableStore = require('obs-store')
...
+ *
+ */
 
 module.exports = class MessageManager extends EventEmitter {
+
+  /**
+   * Controller in charge of managing - storing, adding, removing, updating - Messages.
+   *
+   * @typedef {Object} MessageManager
+   * @param {object} opts @deprecated
```
Comment (bitpshr) [inline app/scripts/lib/message-manager.js]: I know that JSDoc is technically case-insensitive when it comes to types, but I think we should stay consistent throughout (`object` vs. `Object`.)

File: app/scripts/lib/typed-message-manager.js [hunk 4]
```diff
@@ -5,7 +5,38 @@ const assert = require('assert')
 const sigUtil = require('eth-sig-util')
 const log = require('loglevel')
 
+/**
+ * Represents, and contains data about, an 'eth_signTypedData' type signature request. These are created when a
+ * signature for an eth_signTypedData call is requested.
+ *
+ * @see {@link }
```
Comment (bitpshr) [inline app/scripts/lib/typed-message-manager.js]: This link looks like it was accidentally left out. Maybe it's not needed?

**Review by bitpshr — APPROVED** [review]
A few minor nits, but this is looks ready to land otherwise (as soon as CI behaves.) Great info throughout the comments.

**Comment by danjm** [comment]
@bitpshr Your comments were address in 0c26df96

**Comment by danfinlay** [comment]
One comment left, plus tests are failing.

**Pull request merged**